### PR TITLE
Update issueShark modules and realted utils.

### PR DESCRIPTION
**Pull request by myousfi96**

### CODE REVIEW

File: pycoshark/mongomodels.py [hunk 1]
```diff
@@ -729,7 +729,7 @@ class Issue(Document):
     fix_versions = ListField(StringField())
     assignee_id = ObjectIdField()
     issue_links = ListField(DictField())
-    parent_issue_id = ObjectIdField()
+    parent_issue_id = StringField()
```
Comment (atrautsch) [inline pycoshark/mongomodels.py:732]: Why this change? The _id should point to a Issue.id which would make ObjectIdField more suitable.

File: pycoshark/mongomodels.py [hunk 2]
```diff
@@ -779,7 +779,7 @@ class IssueEvent(Document):
     created_at = DateTimeField()
     status = StringField(max_length=50)
     author_id = ObjectIdField()
-    commit_id = ObjectIdField()
+    commit_id = StringField()
```
Comment (atrautsch) [inline pycoshark/mongomodels.py:782]: Same as above

**Review by atrautsch — CHANGES_REQUESTED** [review]

File: pycoshark/mongomodels.py (hunk 1 above)
Comment (myousfi96) [reply pycoshark/mongomodels.py:732]: Because we are now storing the issue external id not the mongo id. I can change the name to reflect the value or we can discuss how we can store the mongo id in the new system

File: pycoshark/mongomodels.py (hunk 2 above)
Comment (myousfi96) [reply pycoshark/mongomodels.py:782]: I'm now storing the commit sha in commit id. I'm planing to revisit all the modules that has commit_id field in it after we finish vcsShark and update the actual commit module.

File: pycoshark/mongomodels.py (hunk 1 above)
Comment (atrautsch) [reply pycoshark/mongomodels.py:732]: Then I would rename it. The parent_issue_id would make me expect a collection unique identifier analogous to issue_id which is not necessarily the case for external_id. I think we can call it parent_issue_external_id to keep it consistent.

File: pycoshark/mongomodels.py (hunk 2 above)
Comment (atrautsch) [reply pycoshark/mongomodels.py:782]: Then it should be commit_sha not commit_id.

**Review by atrautsch — CHANGES_REQUESTED** [review]

File: pycoshark/mongomodels.py (hunk 2 above)
Comment (myousfi96) [reply pycoshark/mongomodels.py]: done

File: pycoshark/mongomodels.py (hunk 1 above)
Comment (myousfi96) [reply pycoshark/mongomodels.py]: I added link_issue function to IssueShark so now we can store issued_id. I put the field type as Dynamic field because first we store the external id an then in link_issue we replace it with object id.

File: pycoshark/utils.py [hunk 3]
```diff
@@ -823,6 +823,16 @@ def delete_last_system_data_on_failure(
...
                 db['action_workflow'].delete_one({'_id': workflow['_id']})
 
+    elif 'issue_system' in system:
+        for issue in db['issue'].find({'issue_system_ids': last_system_id}):
+            if len(issue['issue_system_ids']) > 1:
+                issue['issue_system_ids'].remove(last_system_id)
+                db['issue'].update_one({'_id': issue['_id']}, {'$set': issue}, upsert=False)
+            else:
+                db['issue_comment'].delete_many({'issue_id': issue['_id']})
+                db['issue_event'].delete_one({'_id': issue['_id']})
```
Comment (atrautsch) [inline pycoshark/utils.py:833]: Why only delete one? Could there not be more events for one issue_id?

**Review by atrautsch — CHANGES_REQUESTED** [review]
